code/datums/weather/weather.dm

```diff
@@ -288,3 +392,9 @@
 		gen_overlay_cache += new_weather_overlay
 
 	return gen_overlay_cache
+
+/// Updates the currentpart with the subsystem task that is next in line
+/datum/weather/proc/next_subsystem_task()
+	// loops back to the start of the list once it reaches the end
+	var/next_part = currentpart % length(subsystem_tasks) + 1
+	currentpart = subsystem_tasks[next_part]
```

Comment:
should just use WRAP(), saves you the comment

Reply:
also, the name implies it's a getter, not a proc that assumes it's called in a particular context that then also sets a var. should rename it, or just return the list access, makes things a bit less musty (low priority concern)

Reply:
I was looking for a proc that would do this but I couldn't find anything. I'm going to delete the `next_subsystem_task()` and just use the macro instead.

Reply:
Replaced with `WRAP_UP` via https://github.com/tgstation/tgstation/pull/89550/commits/a793e7d9fc129a90941e5127330fb1334d5fb4a1